### PR TITLE
Handler TraceWriter fix (#2519)

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-functions-host/issues/2519

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Handlers/SystemTraceHandler.cs [hunk 1]
```diff
@@ -10,32 +10,26 @@
...
         {
             _config = config;
         }
 
-        private TraceWriter TraceWriter
+        internal TraceWriter TraceWriter
         {
             get
             {
-                if (_traceWriter == null)
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Handlers/SystemTraceHandler.cs]: Previously we were caching the TraceWriter early on startup, meaning after we apply host.json settings, we never use the updated/configured TraceWriter. In all our controllers for example, the TraceWriter is pulled from DI on each invocation. That's what we do here now.

File: test/WebJobs.Script.Tests/Handlers/SystemTraceHandlerTests.cs [hunk 2]
```diff
@@ -85,5 +87,15 @@ public void SetRequestId_SetsExpectedValue()
             requestId = request.GetRequestId();
             Guid.Parse(requestId);
         }
+
+        [Fact]
+        public void TraceWriter_ReturnsExpectedValue()
+        {
+            var instance = _handler.TraceWriter;
+            instance = _handler.TraceWriter;
+            instance = _handler.TraceWriter;
```
Comment (cgillum) [inline test/WebJobs.Script.Tests/Handlers/SystemTraceHandlerTests.cs]: Why are these lines duplicated?  If this is intentional, can you add a comment explaining what this test is doing?

File: src/WebJobs.Script.WebHost/Handlers/SystemTraceHandler.cs (hunk 1 above)
Comment (cgillum) [reply src/WebJobs.Script.WebHost/Handlers/SystemTraceHandler.cs]: Can I interpret your comment to mean that you're making this trace writing consistent with other parts of the code-base?

File: src/WebJobs.Script.WebHost/Handlers/ExceptionProcessingHandler.cs [hunk 3]
```diff
@@ -24,7 +23,6 @@ public class ExceptionProcessingHandler : ExceptionHandler
     {
         private readonly IDictionary<Type, ExceptionProcessor> _handlers;
         private readonly System.Web.Http.HttpConfiguration _config;
-        private readonly Lazy<TraceWriter> _traceWriterLoader;
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Handlers/ExceptionProcessingHandler.cs]: Realized we had the same issue for our other handler as well

**Pull request merged**